### PR TITLE
Add legacy filtering mode

### PR DESCRIPTION
**Pull request by prasannavl**
/kind feature

- This completes the missing legacy filtering impl to make sure the default output is compatible with previous versions. 
- all / live / gov / <prefix> can be used for the rest.

### CODE REVIEW

File: test/functional/feature_setgov.py [hunk 1]
```diff
@@ -619,8 +619,8 @@ def run_test(self):
         # live test. No live on this. For now, just ensure, no other attrs are listed
         assert_equal(self.nodes[0].listgovs("live"), [[{'ATTRIBUTES': {}}]])
 
-        result_all = self.nodes[0].listgovs("all")
-        result_legacy = self.nodes[0].listgovs("legacy")
+        result_all = self.nodes[0].listgovs("attrs")
```
Comment (prasannavl) [inline test/functional/feature_setgov.py:622]: This is fail, however.

Comment (prasannavl) [reply test/functional/feature_setgov.py:622]: Keep this all.

**Comment by prasannavl** [comment]
Once "attrs" are used. It changes the index. Need to adapt.

**Comment by prasannavl** [comment]
The change appear to have been revert. Added the correct fix, which is to change the index. Also change defaults and added a bit more.

**Review by Bushstar — APPROVED** [review]

**Pull request merged**